velox/functions/lib/Slice.cpp

```diff
@@ -200,22 +211,30 @@ class SliceFunction : public exec::VectorFunction {
     return std::min(endIndex - start, length);
   }
 };
+} // namespace
 
-static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
-  return {// array(T, bigint, bigint) -> array(T)
-          exec::FunctionSignatureBuilder()
-              .typeVariable("T")
-              .returnType("array(T)")
-              .argumentType("array(T)")
-              .argumentType("bigint")
-              .argumentType("bigint")
-              .build()};
+// @param kind The type kind of start and length.
+void registerSliceFunction(const std::string& prefix, TypeKind kind) {
```

Comment:
kind could be a template parameter, and be passed from the registrations of Presto and Spark.

Reply:
Changed it to align mbasmanova's advice https://github.com/facebookincubator/velox/pull/9620#discussion_r1617546384